traffic_ops/testing/api/v4/coordinates_test.go

```diff
@@ -198,3 +238,174 @@ func DeleteTestCoordinates(t *testing.T) {
+		}
+	} else {
+		t.Errorf("No Coordinates found to check pagination")
+	}
+
+	qparams = url.Values{}
+	qparams.Set("limit", "-2")
+	_, _, err = TOSession.GetCoordinates(qparams, nil)
+	if err == nil {
+		t.Error("expected GET Coordinates to return an error when limit is not bigger than -1")
+	} else if !strings.Contains(err.Error(), "must be bigger than -1") {
+		t.Errorf("expected GET Coordinates to return an error for limit is not bigger than -1, actual error: " + err.Error())
+	}
+
+	qparams = url.Values{}
+	qparams.Set("limit", "1")
+	qparams.Set("offset", "0")
+	_, _, err = TOSession.GetCoordinates(qparams, nil)
+	if err == nil {
+		t.Error("expected GET Coordinates to return an error when offset is not a positive integer")
+	} else if !strings.Contains(err.Error(), "must be a positive integer") {
+		t.Errorf("expected GET Coordinates to return an error for offset is not a positive integer, actual error: " + err.Error())
+	}
+
+	qparams = url.Values{}
+	qparams.Set("limit", "1")
+	qparams.Set("page", "0")
+	_, _, err = TOSession.GetCoordinates(qparams, nil)
+	if err == nil {
+		t.Error("expected GET Coordinates to return an error when page is not a positive integer")
+	} else if !strings.Contains(err.Error(), "must be a positive integer") {
+		t.Errorf("expected GET Coordinates to return an error for page is not a positive integer, actual error: " + err.Error())
+	}
+}
+
+func CreateTestCoordinatesWithInvalidName(t *testing.T) {
+	firstCoordinates := testData.Coordinates[0]
+	firstCoordinates.Name = ""
+	_, reqInf, err := TOSession.CreateCoordinate(firstCoordinates)
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Fatalf("Expected 400 status code, got %v", reqInf.StatusCode)
+	}
+	if err == nil {
+		t.Errorf("Getting Coordinates by Invalid Name %v", err)
+	}
+}
+
+func CreateTestCoordinatesWithInvalidLatitude(t *testing.T) {
+	firstCoordinates := testData.Coordinates[0]
+	firstCoordinates.Latitude = 20000
+	_, reqInf, err := TOSession.CreateCoordinate(firstCoordinates)
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Fatalf("Expected 400 status code, got %v", reqInf.StatusCode)
+	}
+	if err == nil {
+		t.Errorf("Getting Coordinates by Invalid Latitude %v", err)
+	}
+}
+
+func CreateTestCoordinatesWithInvalidLogitude(t *testing.T) {
+	firstCoordinates := testData.Coordinates[0]
+	firstCoordinates.Longitude = 20000
+	_, reqInf, err := TOSession.CreateCoordinate(firstCoordinates)
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Fatalf("Expected 400 status code, got %v", reqInf.StatusCode)
+	}
+	if err == nil {
+		t.Errorf("Getting Coordinates by Invalid Longitude %v", err)
+	}
+}
+
+func UpdateTestCoordinatesByInvalidId(t *testing.T) {
+	firstCoord := testData.Coordinates[0]
+	resp, reqInf, err := TOSession.GetCoordinateByName(firstCoord.Name, nil)
+	if err != nil {
+		t.Errorf("cannot GET Coordinate by name: %v - %v", firstCoord.Name, err)
+	}
+	coord := resp[0]
```

Comment:
this will segfault if Traffic Ops returns an empty response, which can occur if the creation failed for this Coordinate or if there's simply something wrong with the GET handler

Reply:
added null checks